extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/controlBar.js

```diff
@@ -0,0 +1,506 @@
+
+      "quality": (
+        <div className="oo-popover-button-container" key="quality">
+          {videoQualityPopover}
+          <a className={qualityClass} onClick={this.handleQualityClick} style={selectedStyle}>
+            <Icon {...this.props} icon="quality" style={dynamicStyles.iconCharacter}
+              onMouseOver={this.highlight} onMouseOut={this.removeHighlight}/>
+          </a>
+        </div>
+      ),
+
+      "discovery": <a className="oo-discovery oo-control-bar-item"
+        onClick={this.handleDiscoveryClick} key="discovery">
+        <Icon {...this.props} icon="discovery" style={dynamicStyles.iconCharacter}
+          onMouseOver={this.highlight} onMouseOut={this.removeHighlight}/>
+      </a>,
+
+      "closedCaption": (
+        <div className="oo-popover-button-container" key="closedCaption">
+          {closedCaptionPopover}
+          <a className={captionClass} onClick={this.handleClosedCaptionClick} style={selectedStyle}>
+            <Icon {...this.props} icon="cc" style={dynamicStyles.iconCharacter}
+              onMouseOver={this.highlight} onMouseOut={this.removeHighlight}/>
+          </a>
+        </div>
+      ),
+
+      "share": <a className="oo-share oo-control-bar-item"
+        onClick={this.handleShareClick} key="share">
+        <Icon {...this.props} icon="share" style={dynamicStyles.iconCharacter}
+          onMouseOver={this.highlight} onMouseOut={this.removeHighlight}/>
+      </a>,
+
+      "fullscreen": <a className="oo-fullscreen oo-control-bar-item"
+        onClick={this.handleFullscreenClick} key="fullscreen">
+        <Icon {...this.props} icon={fullscreenIcon} style={dynamicStyles.iconCharacter}
+          onMouseOver={this.highlight} onMouseOut={this.removeHighlight}/>
+      </a>,
+
+      "logo": <Logo key="logo" imageUrl={this.props.skinConfig.controlBar.logo.imageResource.url}
+        clickUrl={this.props.skinConfig.controlBar.logo.clickUrl}
+        target={this.props.skinConfig.controlBar.logo.target}
+        width={this.props.responsiveView != this.props.skinConfig.responsive.breakpoints.xs.id ? this.props.skinConfig.controlBar.logo.width : null}
+        height={this.props.skinConfig.controlBar.logo.height}/>
+    };
+
+    var controlBarItems = [];
+    var defaultItems = this.props.controller.state.isPlayingAd ? this.props.skinConfig.buttons.desktopAd : this.props.skinConfig.buttons.desktopContent;
+
+    //if mobile and not showing the slider or the icon, extra space can be added to control bar width. If volume bar is shown instead of slider, add some space as well:
+    var volumeItem = null;
+    var extraSpaceVolume = 0;
+
+    for (var j = 0; j < defaultItems.length; j++) {
+      if (defaultItems[j].name == "volume") {
+        volumeItem = defaultItems[j];
+
+        var extraSpaceVolumeSlider = (((volumeItem && this.isMobile && !this.props.controller.state.volumeState.volumeSliderVisible) || volumeItem && Utils.isIos()) ? parseInt(volumeItem.minWidth) : 0);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)